### PR TITLE
Feat: Change port name delimiter from _ to |

### PR DESCRIPTION
**Pull request by 01Parzival10**

### CODE REVIEW

**Comment by 01Parzival10** [comment]
I hope thats everything

**Comment by uuqjz** [comment]
If you look at #193 we wanted to use a pipe as delimiter

**Comment by uuqjz** [comment]
I don't quite understand your regex changes. You put `;`behind some `_`but not all. Additionally, shouldn't the new delimiter replace the old one?

**Comment by 01Parzival10** [comment]
> I don't quite understand your regex changes. You put `;`behind some `_`but not all. Additionally, shouldn't the new delimiter replace the old one?

You do want flow names to be allowed to have "_" in them, right?
| are only allowed in flow names, so in the latter part of the Regex

**Comment by uuqjz** [comment]
> > I don't quite understand your regex changes. You put `;`behind some `_`but not all. Additionally, shouldn't the new delimiter replace the old one?
> 
> You do want flow names to be allowed to have "_" in them, right? | are only allowed in flow names, so in the latter part of the Regex

You mean | should only be allowed between flow names, right?

**Comment by 01Parzival10** [comment]
> > > I don't quite understand your regex changes. You put `;`behind some `_`but not all. Additionally, shouldn't the new delimiter replace the old one?
> > 
> > You do want flow names to be allowed to have "_" in them, right? | are only allowed in flow names, so in the latter part of the Regex
> 
> You mean | should only be allowed between flow names, right?

Yes. 

But you do want "_" to be allowed in a single flow name and also in labels to be compatible with MicroSecEnd? That's why I didn't remove it from the Regex

**Comment by uuqjz** [comment]
> But you do want "_" to be allowed in a single flow name and also in labels to be compatible with MicroSecEnd? That's why I didn't remove it from the Regex
Yes id like that

**Comment by 01Parzival10** [comment]
> > But you do want "_" to be allowed in a single flow name and also in labels to be compatible with MicroSecEnd? That's why I didn't remove it from the Regex
> 
> Yes id like that

Alright, it should be ready to merge then.

**Comment by uuqjz** [comment]
> > > But you do want "_" to be allowed in a single flow name and also in labels to be compatible with MicroSecEnd? That's why I didn't remove it from the Regex
> > 
> > 
> > Yes id like that
> 
> Alright, it should be ready to merge then.

I will test the regex some more tomorrow then merge if i don't find an edgecase

File: src/features/dfdElements/outputPortBehaviorValidation.ts [hunk 1]
```diff
@@ -23,13 +23,13 @@ export class PortBehaviorValidator {
     // Regex that validates a set statement.
     // Has the label type and label value that should be set as capturing groups.
     private static readonly SET_REGEX =
-        /^set +([A-Za-z][A-Za-z0-9_]*)\.([A-Za-z][A-Za-z0-9_]*) *= *(?: +|!|TRUE|FALSE|\|\||&&|\(|\)|[A-Za-z][A-Za-z0-9_]*(?:\.[A-Za-z][A-Za-z0-9_]*){2})+$/;
+        /^set +([A-Za-z][A-Za-z0-9_]*)\.([A-Za-z][A-Za-z0-9_]*) *= *(?: +|!|TRUE|FALSE|\|\||&&|\(|\)|[A-Za-z][A-Za-z0-9_\|]*(?:\.[A-Za-z][A-Za-z0-9_\|]*){2})+$/;
```
Comment (uuqjz) [inline src/features/dfdElements/outputPortBehaviorValidation.ts:26]: The regex matches `set sensitivity.private = auth|server.sensi|tivity.priv|ate`
The last to pipes have to be remove as we don't want to allow pipes in labels

File: src/features/dfdElements/outputPortBehaviorValidation.ts [hunk 2]
```diff
@@ -23,13 +23,13 @@ export class PortBehaviorValidator {
...
     // Regex that is used to extract all inputs, their label types and label values from a set statement.
     // Each input is a match with the input name, label type and label value as capturing groups.
     private static readonly SET_REGEX_EXPRESSION_INPUTS =
-        /([A-Za-z][A-Za-z0-9_]*)\.([A-Za-z][A-Za-z0-9_]*)\.([A-Za-z][A-Za-z0-9_]*)/g;
+        /([A-Za-z][A-Za-z0-9_\|]*)\.([A-Za-z][A-Za-z0-9_\|]*)\.([A-Za-z][A-Za-z0-9_\|]*)/g;
```
Comment (uuqjz) [inline src/features/dfdElements/outputPortBehaviorValidation.ts:30]: Same here type and value should not contain pipes

**Review by uuqjz — CHANGES_REQUESTED** [review]
Please adopt the two comments

File: src/features/dfdElements/outputPortBehaviorValidation.ts (hunk 2 above)
Comment (01Parzival10) [reply src/features/dfdElements/outputPortBehaviorValidation.ts]: Done

**Review by uuqjz — APPROVED** [review]
LGTM

**Pull request merged**